cmd/earthly/buildkit.go

```diff
@@ -178,6 +179,28 @@ func (app *earthlyApp) configureSatellite(cliCtx *cli.Context, cloudClient cloud
 	}
 	app.buildkitdSettings.SatelliteToken = token
 
+	// Reserve the satellite for the upcoming build.
+	// This operation can take a moment if the satellite is asleep.
+	console := app.console.WithPrefix("satellite")
+	out := make(chan string)
+	var reserveErr error
+	go func() { reserveErr = cloudClient.ReserveSatellite(cliCtx.Context, app.satelliteName, orgID, out) }()
+	for status := range out {
+		switch status {
+		case cloud.SatelliteStatusSleep:
+			console.Printf("%s is waking up. Please wait...", app.satelliteName)
+		case cloud.SatelliteStatusStarting:
+			console.VerbosePrintf("...Still waking up...")
```

Comment:
very cool!

Maybe we can be silly and do something like:

```
wakeupText := []string{
    "tracking orbit",
    "adjusting course",
    "modifying stasis chamber temperature",
    ...
   }
console.VerbosePrintf("...%s...", wakeupText[rand.Intn(len(wakeupText))])
```

Reply:
Hehe, I love that actually. Especially for when we port this kind of API to the Launch and Delete endpoints

Reply:
LoL hilarious :D love it

Reply:
amazing list of statuses :clap: